rotkehlchen/chain/optimism/modules/velodrome/velodrome_cache.py

```diff
@@ -0,0 +1,218 @@
+                address=pool.gauge_address,
+                name=f'Velodrome gauge for Velodrome pool {pool.pool_name}',
+                blockchain=SupportedBlockchain.OPTIMISM,
+            ))
+        try:
+            db_addressbook.add_addressbook_entries(
+                write_cursor=write_cursor,
+                entries=addresbook_entries,
+            )
+        except InputError as e:
+            log.debug(
+                f'Velodrome address book names for pool {pool.pool_address} were not added. '
+                f'Probably names were added by the user earlier. {e!s}')
+
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.VELODROME_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        if pool.gauge_address is not None:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.VELODROME_GAUGE_ADDRESS, pool.pool_address),
+                value=pool.gauge_address,
+            )
+
+
+def read_velodrome_pools_and_gauges_from_cache() -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:  # noqa: E501
+    """
+    Reads globaldb cache and returns:
+    - A set of all known addresses of velodrome pools.
+    - A set of all known addresses of velodrome gauges.
+
+    Doesn't raise anything unless cache entries were inserted incorrectly.
+    """
+    gauge_addresses = set()
+    pool_addresses = set()
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        str_pool_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(CacheType.VELODROME_POOL_ADDRESS,))  # noqa: E501
+        for pool_address in str_pool_addresses:
+            gauge_address = globaldb_get_unique_cache_value(cursor=cursor, key_parts=(CacheType.VELODROME_GAUGE_ADDRESS, pool_address))  # noqa: E501
+            if gauge_address is None:
+                continue
+            gauge_addresses.add(string_to_evm_address(gauge_address))
+            pool_addresses.add(string_to_evm_address(pool_address))
+
+    return pool_addresses, gauge_addresses
+
+
+def query_velodrome_pools_from_chain(
+        inquirer: 'OptimismInquirer',
+        existing_pools: list[ChecksumEvmAddress],
+) -> Optional[list[VelodromePoolData]]:
+    """
+    Queries velodrome data from chain from the Velodrome Finance LP Sugar v2 contract
+    (Find more details for the contract here: https://github.com/velodrome-finance/sugar).
+
+    May raise:
+    - RemoteError if there is an error connecting with the remote source of data
+    """
+    data_contract = inquirer.contracts.contract(VELODROME_SUGAR_V2_CONTRACT)
+    pool_data = data_contract.call(
+        node_inquirer=inquirer,
+        method_name='all',
+        arguments=[1000, 0, '0x48a264d1BC1765598D327bB60789bEd541b12504'],  # limit, offset, address  # no need to use a specific address here we just want the pool data  # noqa: E501
+    )
```

Comment:
Maybe you should have a lower limit and loop trough the responses. As you have it now there are potential pools that you are missing. Also I'm not sure but I believe it can happen that the query fails depending on the configuration for the max gas used in the queried node.

I would change this to having a lower limit and playing with the offset. Also can you explain that the `address` is something from the design of this interface and to cover one of the use cases that the developers had in mind and that we don't need it since all the pools are returned